ddtrace/_trace/processor/__init__.py

```diff
@@ -321,16 +335,29 @@ def on_span_finish(self, span):
                     finished = trace_spans
 
                 num_finished = len(finished)
+                if num_finished != trace.num_finished:
+                    log_msg = f"Finished span count of {num_finished} is not the expected {trace.num_finished}. {span}"
+                    if config._telemetry_enabled:
+                        telemetry.telemetry_writer.add_log("WARNING", log_msg)
+                    log.warning(log_msg)
```

Comment:
Very, very nit-y (and very "6 vs half dozen"-y) : I think I like this better, because it explicitly shows that we **should** be at 0 after we remove the finished spans, and makes the forcing of `trace.num_finished=0` a result of that condition:  
```suggestion
                trace.num_finished -= num_finished           
                if trace.num_finished != 0:
                    log_msg = f"Finished span count of {num_finished} is not the expected {trace.num_finished}. {span}"
                    if config._telemetry_enabled:
                        telemetry.telemetry_writer.add_log("WARNING", log_msg)
                    log.warning(log_msg)
                    trace.num_finished = 0
```
Also because there are deleted lines It didn't let me suggest removing the next 3 lines.

Reply:
(since this is a nit, feel free to resolve)